pkg/webhook/registration.go

```diff
@@ -0,0 +1,229 @@
+		for _, t := range webhook.Types {
+			rule, err := buildRule(mgr, t)
+			if err != nil {
+				return err
+			}
+			rules = append(rules, *rule)
+		}
+
+		webhookToRegister = append(webhookToRegister, admissionregistrationv1beta1.Webhook{
+			Name:              fmt.Sprintf("%s.%s.extensions.gardener.cloud", webhook.Name, strings.TrimPrefix(providerName, "provider-")),
+			NamespaceSelector: namespaceSelector,
+			FailurePolicy:     &fail,
+			ClientConfig:      buildClientConfigFor(webhook, namespace, providerName, port, caBundle),
+			Rules:             rules,
+		})
+	}
+
+	s := runtime.NewScheme()
+	if err := scheme.AddToScheme(s); err != nil {
+		return err
+	}
+
+	c, err := client.New(mgr.GetConfig(), client.Options{Scheme: s})
+	if err != nil {
+		return err
+	}
+
+	_, err = controllerutil.CreateOrUpdate(ctx, c, mutatingWebhookConfiguration, func() error {
+		mutatingWebhookConfiguration.Webhooks = webhookToRegister
+		return nil
+	})
+	return err
+}
+
+// buildSelector creates and returns a LabelSelector for the given webhook kind and provider.
+func buildSelector(kind, provider string) (*metav1.LabelSelector, error) {
+	// Determine label selector key from the kind
+	var key string
+	switch kind {
+	case SeedKind:
+		key = gardencorev1alpha1.SeedProvider
+	case ShootKind:
+		key = gardencorev1alpha1.ShootProvider
+	case BackupKind:
+		key = gardencorev1alpha1.BackupProvider
+	default:
+		return nil, fmt.Errorf("invalid webhook kind '%s'", kind)
+	}
+
+	// Create and return LabelSelector
+	return &metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{Key: key, Operator: metav1.LabelSelectorOpIn, Values: []string{provider}},
+		},
+	}, nil
+}
+
+// buildRule creates and returns a RuleWithOperations for the given object type.
+func buildRule(mgr manager.Manager, t runtime.Object) (*admissionregistrationv1beta1.RuleWithOperations, error) {
+	// Get GVK from the type
+	gvk, err := apiutil.GVKForObject(t, mgr.GetScheme())
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not get GroupVersionKind from object %v", t)
+	}
+
+	// Get REST mapping from GVK
+	mapping, err := mgr.GetRESTMapper().RESTMapping(gvk.GroupKind(), gvk.Version)
+	if err != nil {
+		return nil, errors.Wrapf(err, "could not get REST mapping from GroupVersionKind '%s'", gvk.String())
+	}
+
+	// Create and return RuleWithOperations
+	return &admissionregistrationv1beta1.RuleWithOperations{
+		Operations: []admissionregistrationv1beta1.OperationType{
+			admissionregistrationv1beta1.Create,
+			admissionregistrationv1beta1.Update,
+		},
+		Rule: admissionregistrationv1beta1.Rule{
+			APIGroups:   []string{gvk.Group},
+			APIVersions: []string{gvk.Version},
+			Resources:   []string{mapping.Resource.Resource},
+		},
+	}, nil
+}
+
+func buildClientConfigFor(webhook *Webhook, namespace, providerName string, port int, caBundle []byte) admissionregistrationv1beta1.WebhookClientConfig {
+	path := "/" + webhook.Path
```

Comment:
Maybe use `filepath.Join` ?

Reply:
This is not a file path, it's a URL. Good enough imho.